src/test/java/streams/part1/exercise/Exercise2.java

```diff
@@ -0,0 +1,141 @@
+package streams.part1.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import lombok.Value;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+public class Exercise2 {
+
+    @Test
+    public void calcAverageAgeOfEmployees() {
+        List<Employee> employees = getEmployees();
+
+        Double expected = employees
+                .stream()
+                .map(Employee::getPerson)
+                .collect(Collectors.averagingInt(Person::getAge));
+
+        assertEquals(33.66, expected, 0.1);
+    }
+
+    @Test
+    public void findPersonWithLongestFullName() {
+        List<Employee> employees = getEmployees();
+
+        @Value
+        class PersonIntegerPair {
+            Person person;
+            Integer integer;
+        }
+
+        Person expected = employees
+                .stream()
+                .map(Employee::getPerson)
+                .map(person -> new PersonIntegerPair(person, person.getFullName().length()))
+                .max(Comparator.comparingInt(PersonIntegerPair::getInteger))
+                .map(PersonIntegerPair::getPerson)
+                .orElseThrow(RuntimeException::new);
```

Comment:
Are you sure that you need all this operations?
I guess, it's enough:
1. map to persons,
2. sort the full names in descending order
3. get first.

On the other hand, if you want to use utility class - you must name it (and its fields) logically for using.
Class name ```PersonInteger``` and its field ```Integer integer``` don't give enough information about content.
At this place you can name it like ```PersonFullNameLengthPair``` and ```Integer fullNameLength``` respectively.

Also, it's not good to throw ```RuntimeException```, because it don't give to the reader information about its reason.
For example, you can use ```NoSuchElementException``` instead.